tweet_recommendations/embeddings/sif_embedding.py

```diff
@@ -57,51 +71,74 @@ def fit(self, x: pd.DataFrame, y: Optional[pd.DataFrame] = None,
+            its corresponding embeddings.
         :return: np.ndarray of sentences embeddings.
         """
-        lemmatized = []
-        if isinstance(x, pd.DataFrame):
-            lemmatized = x["lemmas"].to_list()
-        elif isinstance(x[0], str):
-            for i, xi in enumerate(x):
-                lemmatized.append(get_wcrft2_results_for_text(xi))
-        elif isinstance(x[0], list):
-            for i, xi in enumerate(x):
-                lemmatized.append(xi)
-        output = self._get_sif_embedding(lemmatized)
+        if self.keyed_vector_model:
+            lemmatized = []
+            if isinstance(x, pd.DataFrame):
+                lemmatized = x["lemmas"].to_list()
+            elif isinstance(x[0], str):
+                for i, xi in enumerate(x):
+                    lemmatized.append(get_wcrft2_results_for_text(xi))
+            elif isinstance(x[0], list):
+                for i, xi in enumerate(x):
+                    lemmatized.append(xi)
+            data = lemmatized
+        else:
+            data = x
+        output = self._get_sif_embedding(data)
         return np.asarray(output)
 
-    def _get_weighted_average_embeddings(self, sentences: List[List[str]]) -> np.ndarray:
+    def _get_weighted_average_embeddings(self, data: pd.DataFrame) -> np.ndarray:
         """
         Compute the weighted average vectors for given tweet content lemmas.
-        :param sentences: List of tweet content lemmas list.
+        :param data: DataFrame which should contain tweet text lemmas.
         :return: Weighted average of tweet embeddings.
         """
+        sentences = data["lemmas"].values.tolist()
         n_samples = len(sentences)
-        emb = np.zeros((n_samples, self.keyed_vector_model.vector_size))
+        if self.keyed_vector_model:
+            embedding_size = self.keyed_vector_model.vector_size
+        else:
+            embedding_size = len(data[self.embedding_name].iloc[0])
+        emb = np.zeros((n_samples, embedding_size))
+
         for i in tqdm.trange(n_samples, disable=not self.verbose):
             if sentences[i]:
                 words_embeddings = []
                 words_weights = []
                 for word in sentences[i]:
                     try:
-                        word_embedding = self.keyed_vector_model.word_vec(word.lower())
-                        words_embeddings.append(word_embedding)
+                        if self.keyed_vector_model:
+                            word_embedding = self.keyed_vector_model.word_vec(word.lower())
+                            words_embeddings.append(word_embedding)
                         words_weights.append(self.words_weights.get(word.lower(), 0))
                     except KeyError:
                         continue
+                if not self.keyed_vector_model:
+                    words_embeddings = data["word_embeddings"].iloc[i]
                 sentence_word_embeddings = np.array(words_embeddings)
```

Comment:
podejrzewam, że w tym miejscu puchnie Ci zajętość pamięci. `np.array(a)` kopiuje wartości `a`, zamiast tego lepiej użyć `np.asarray(a)`